static/app/views/dashboardsV2/title.tsx

```diff
@@ -1,87 +1,42 @@
 import React from 'react';
 import styled from '@emotion/styled';
 
-import {addErrorMessage} from 'app/actionCreators/indicator';
-import InlineInput from 'app/components/inputInline';
+import EditableText from 'app/components/editableText';
 import {t} from 'app/locale';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
 import space from 'app/styles/space';
 
 import {DashboardDetails} from './types';
 
 type Props = {
   dashboard: DashboardDetails | null;
-  isEditing: boolean;
   onUpdate: (dashboard: DashboardDetails) => void;
 };
 
-class DashboardTitle extends React.Component<Props> {
-  refInput = React.createRef<InlineInput>();
-
-  onBlur = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const nextDashboardTitle = (event.target.value || '').trim().slice(0, 255).trim();
-
-    if (!nextDashboardTitle) {
-      addErrorMessage(t('Please set the title for this dashboard'));
-
-      // Help our users re-focus so they cannot run away from this problem
-      if (this.refInput.current) {
-        this.refInput.current.focus();
-      }
-
-      return;
-    }
-    const {dashboard, onUpdate} = this.props;
-    if (!dashboard) {
-      return;
-    }
-
-    event.target.innerText = nextDashboardTitle;
-
-    onUpdate({
-      ...dashboard,
-      title: nextDashboardTitle,
-    });
-  };
-
-  render() {
-    const {dashboard, isEditing} = this.props;
-
-    if (!dashboard) {
-      return <Container>{t('Dashboards')}</Container>;
-    }
-
-    if (!isEditing) {
-      return <Container>{dashboard.title}</Container>;
-    }
-
-    return (
-      <Container>
-        <StyledInlineInput
-          name="dashboard-title"
-          ref={this.refInput}
+function DashboardTitle({dashboard, onUpdate}: Props) {
+  return (
+    <Container>
+      {!dashboard ? (
+        t('Dashboards')
+      ) : (
+        <EditableText
           value={dashboard.title}
-          onBlur={this.onBlur}
+          onChange={newTitle => onUpdate({...dashboard, title: newTitle})}
+          errorMessage={t('Please set a title for this widget')}
+          successMessage={t('Widget title saved successfully')}
```

Comment:
Isn't this the title for the entire dashboard?

Reply:
good catch :)